### PR TITLE
Add transfer log check for transaction

### PR DESCRIPTION
**Pull request by paologalligit**
This refers to this [ticket](https://github.com/vechain/protocol-board-repo/issues/252).

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -9,5 +9,8 @@
         "resolveJsonModule": true
     },
     "include": ["./scripts", "./test", "./typechain-types"],
-    "files": ["./hardhat.config.ts"]
+    "files": [
+        "./hardhat.config.ts",
+        "./test/thorest-api/transactions/newFile.ts"
```
Comment (leszek-vechain) [inline tsconfig.json:14]: do we need this newfile for anything ?

File: test/thorest-api/transactions/setup/transaction-data-driven-flow.ts [hunk 2]
```diff
@@ -24,72 +22,123 @@ class TransactionDataDrivenFlow {
...
 
-export {
-    TransactionDataDrivenFlow
+function createDefaultLogFilterRequest(
+    blockNumber: number | undefined
+): components['schemas']['TransferLogFilterRequest'] {
+    return {
+        range: {
+            to: blockNumber,
+            from: blockNumber,
```
Comment (leszek-vechain) [inline test/thorest-api/transactions/setup/transaction-data-driven-flow.ts:136]: should we have block range or just block - block is enough ?

File: tsconfig.json (hunk 1 above)
Comment (paologalligit) [reply tsconfig.json:14]: ops, good catch! That's garbage

File: test/thorest-api/transactions/setup/transaction-data-driven-flow.ts (hunk 2 above)
Comment (paologalligit) [reply test/thorest-api/transactions/setup/transaction-data-driven-flow.ts:136]: This is something that can be used as a default log filter: it looks for the block the tx is added into, but if you want to have a customized filter you can pass it directly when creating the testPlan in the transaction log step

**Review by leszek-vechain — APPROVED** [review]

**Pull request merged**